crates/flashblocks-rpc/src/rpc.rs

```diff
@@ -387,7 +395,41 @@ where
                         block_number: Some(block_number),
                         base_fee,
                     };
-                    Ok(Some(self.transform_tx(transaction, tx_info)))
+                    // preload transaction receipt if it's a deposit transaction
+                    if transaction.is_deposit() {
+                        let receipt =
+                            EthTransactions::transaction_receipt(&self.eth_api, tx_hash).await;
+
+                        match receipt {
+                            Ok(Some(txn_receipt)) => {
+                                let envelope: OpReceiptEnvelope = txn_receipt.into();
+
+                                if let OpReceiptEnvelope::Deposit(deposit_receipt) = envelope {
+                                    return Ok(Some(self.transform_tx(
+                                        transaction,
+                                        tx_info,
+                                        Some(deposit_receipt.receipt),
+                                    )));
+                                }
+                            }
+                            // Ok(None) or Err(e)
+                            Ok(None) => {
+                                error!(
+                                    "could not find receipt for block transaction: {:?}",
+                                    tx_hash
+                                );
+                                return Ok(None);
+                            }
+                            Err(e) => {
+                                error!(
+                                    "unable to fetch receipt for block transaction: {:?} error: {:?}",
+                                    tx_hash, e
+                                );
+                                return Err(e.into());
```

Comment:
You can remove this if you just use a `?;` on line 400

Reply:
doesn't that get rid of the error log we wish to have? also, im getting some trait errors if i try to do that. the type we get back from EthAPI doesn't have the `From` trait implemented to implicitly convert it